### PR TITLE
fix: better AB import error

### LINKED ISSUE #889: [Android] - Uploading a csv in the AB errors out and displays [Object object]
## Bug description
Trying to upload a csv in the Address book doesn't work.
Tested it in Android

## Environment
 - Browser: Chrome

## Steps to reproduce
 1. Need to have downloaded a csv in your phone already. Export csv feature works fine, you can use that
 2. Try to upload the csv

## Expected result
The AB gets updated

## Obtained result
An error is displayed.

### PR DESCRIPTION
**Pull request by DiogoSoaress**
## What it solves
Resolves #889

## How this PR fixes it
Removes the extraneous content type validation.
Adds a `papaparse` error type.

## How to test it
Follow #[889](https://github.com/safe-global/web-core/issues/889) steps

### CODE REVIEW

File: src/components/address-book/ImportDialog/index.tsx [hunk 1]
```diff
@@ -16,9 +16,18 @@ import css from './styles.module.css'
 import { trackEvent, ADDRESS_BOOK_EVENTS } from '@/services/analytics'
 import { abCsvReaderValidator, abOnUploadValidator } from './validation'
 import ErrorMessage from '@/components/tx/ErrorMessage'
+import { Errors, logError } from '@/services/exceptions'
 
 type AddressBookCSVRow = ['address', 'name', 'chainId']
 
+// https://react-papaparse.js.org/docs#errors
+type PapaparseErrorType = {
```
Comment (iamacook) [inline src/components/address-book/ImportDialog/index.tsx:24]: This [can also have an (optional) `index` property](https://github.com/mholt/PapaParse/blob/master/papaparse.js#L1524) and [`row` can be `undefined`](https://github.com/mholt/PapaParse/blob/master/papaparse.js#L1378-L1380)

Comment (DiogoSoaress) [reply src/components/address-book/ImportDialog/index.tsx:24]: Thanks for the observations. I've pushed an improved type

**Review by iamacook — APPROVED** [review]

**Comment by francovenica** [comment]
I still have the same issue. I got an error in the console now
image

File: src/components/address-book/ImportDialog/index.tsx [hunk 2]
```diff
@@ -70,15 +80,17 @@ const ImportDialog = ({ handleClose }: { handleClose: () => void }): ReactElemen
...
+          onUploadRejected={(result: { file: File; errors?: Array<Error | string | PapaparseErrorType> }[]) => {
             setZoneHover(false)
             setError(undefined)
 
             // csvReaderValidator error
             const error = result?.[0].errors?.pop()
 
             if (error) {
-              setError(error instanceof Error ? error.message : error.toString())
+              const errorDescription = error instanceof Error ? error.message : JSON.stringify(error)
```
Comment (katspaugh) [inline src/components/address-book/ImportDialog/index.tsx:91]: Stringify adds extra quotes around strings as you can see on Franco’s screenshot.

**Comment by DiogoSoaress** [comment]
Merging this improvement after testing with @francovenica as we improved and log the import csv error.

**Pull request merged**